File: app/src/main/java/it/niedermann/nextcloud/deck/DeckApplication.java
package it.niedermann.nextcloud.deck;

import android.app.Application;
import android.os.StrictMode;

import java.util.logging.Logger;

import it.niedermann.nextcloud.deck.feature.shared.util.CustomAppGlideModule;


public class DeckApplication extends Application {

    private static final Logger logger = Logger.getLogger(DeckApplication.class.getName());

    @Override
    public void onCreate() {

        if (BuildConfig.DEBUG) {
            enableStrictModeLogging();
        }

        super.onCreate();
    }

    @Override
    public void onLowMemory() {
        super.onLowMemory();
        logger.warning("--- Low memory: Clear Glide cache ---");
        CustomAppGlideModule.clearCache(this);
        logger.warning("--- Low memory: Clear debug log ---");
    }

    private void enableStrictModeLogging() {
        StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder()
                .detectAll()
                .permitDiskReads()
                .penaltyLog()
                .build());
        StrictMode.setVmPolicy(new StrictMode.VmPolicy.Builder()
                .detectAll()
                .penaltyLog()
                .build());
    }
}

File: app/android/src/main/java/it/niedermann/nextcloud/deck/DeckApplication.kt
package it.niedermann.nextcloud.deck

import android.app.Application
import android.content.pm.ApplicationInfo
import android.os.StrictMode
import dagger.hilt.android.HiltAndroidApp
import it.niedermann.nextcloud.deck.domain.usecases.users.GetAvatarUseCase
import it.niedermann.nextcloud.deck.ui.components.AvatarProvider
import it.niedermann.nextcloud.deck.ui.exception.ExceptionHandler
import jakarta.inject.Inject

@HiltAndroidApp
class DeckApplication : Application() {

    @Inject
    lateinit var getAvatarUseCase: GetAvatarUseCase

    override fun onCreate() {
        if (applicationInfo.flags and ApplicationInfo.FLAG_DEBUGGABLE != 0) {
            StrictMode.setThreadPolicy(
                StrictMode.ThreadPolicy.Builder()
                    .detectDiskReads()
                    .detectDiskWrites()
                    .detectNetwork()
                    .penaltyLog()
                    .build()
            )
            StrictMode.setVmPolicy(
                StrictMode.VmPolicy.Builder()
                    .detectLeakedSqlLiteObjects()
                    .detectLeakedClosableObjects()
                    .penaltyLog()
                    .build()
            )
        }
        ExceptionHandler.initialize(this)
        super.onCreate()
        AvatarProvider.initialize(getAvatarUseCase)
    }
}
